tests/unit/development/metadata_test.py

```diff
@@ -90,8 +124,37 @@ class Table(Component):
     _base_nodes = ['optionalNode', 'optionalElement', 'children']
     _namespace = 'TableComponents'
     _type = 'Table'
+    OptionalEnumEnum = OptionalEnumEnum
+    OptionalObjectWithExactAndNestedDescriptionFigure = OptionalObjectWithExactAndNestedDescriptionFigure
+    OptionalObjectWithExactAndNestedDescription = OptionalObjectWithExactAndNestedDescription
+    OptionalObjectWithShapeAndNestedDescriptionFigure = OptionalObjectWithShapeAndNestedDescriptionFigure
+    OptionalObjectWithShapeAndNestedDescription = OptionalObjectWithShapeAndNestedDescription
     @_explicitize_args
-    def __init__(self, children=None, optionalArray=Component.UNDEFINED, optionalBool=Component.UNDEFINED, optionalFunc=Component.UNDEFINED, optionalNumber=Component.UNDEFINED, optionalObject=Component.UNDEFINED, optionalString=Component.UNDEFINED, optionalSymbol=Component.UNDEFINED, optionalNode=Component.UNDEFINED, optionalElement=Component.UNDEFINED, optionalMessage=Component.UNDEFINED, optionalEnum=Component.UNDEFINED, optionalUnion=Component.UNDEFINED, optionalArrayOf=Component.UNDEFINED, optionalObjectOf=Component.UNDEFINED, optionalObjectWithExactAndNestedDescription=Component.UNDEFINED, optionalObjectWithShapeAndNestedDescription=Component.UNDEFINED, optionalAny=Component.UNDEFINED, customProp=Component.UNDEFINED, customArrayProp=Component.UNDEFINED, id=Component.UNDEFINED, **kwargs):
+    def __init__(
+        self,
+        children: typing.Union[str, int, float, Component, typing.List[typing.Union[str, int, float, Component]]] = None,
+        optionalArray: typing.List = Component.UNDEFINED,
+        optionalBool: bool = Component.UNDEFINED,
+        optionalFunc: typing.Any = Component.UNDEFINED,
+        optionalNumber: typing.Union[int, float, numbers.Number] = Component.UNDEFINED,
+        optionalObject: dict = Component.UNDEFINED,
+        optionalString: str = Component.UNDEFINED,
+        optionalSymbol: typing.Any = Component.UNDEFINED,
+        optionalNode: typing.Union[str, int, float, Component, typing.List[typing.Union[str, int, float, Component]]] = Component.UNDEFINED,
+        optionalElement: Component = Component.UNDEFINED,
```

Comment:
Huh, never noticed this before... do we actually support `Element` as distinct from `Node`?

Reply:
Yes, it ends up rendering the same as node but for typing it allows only a single component to be put and not a list of string/number/components.

Reply:
Are there any examples of this in existing components? Does it work, either as `children` or with components-as-props?

Reply:
Before component as props, it was kind of invalid prop as only the children would be hydrated and children would be treated as a node, component as props also doesn't differ those Element from node.

The collect_nodes make both node and element into component:
https://github.com/plotly/dash/blob/453834bbd0083d66d73752ca5d7c25b30d3583cc/dash/development/_collect_nodes.py#L1-L2

I think the only other place we have Element defined is in the typescript props:
https://github.com/plotly/dash/blob/453834bbd0083d66d73752ca5d7c25b30d3583cc/@plotly/dash-generator-test-component-typescript/src/props.ts#L31
But the prop itself was not tested as it was before component as props was supported.